src/query/expression/src/kernels/filter.rs

```diff
@@ -190,7 +190,7 @@ impl Column {
         let mut start_index: usize = 0;
 
         if offset > 0 {
-            let n = 8 - offset;
+            let n = std::cmp::min(8 - offset, length);
```

Comment:
Need a comment, I did not know the reason.

Reply:
More comments need for the changes if we think this is issue before. Make the code  more readable for human.